src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -95,6 +99,33 @@ public RoomDto startGame(String roomId, StartGameRequest request) {
         return roomDto;
     }
 
+    public RoomDto playAgain(String roomId, PlayAgainRequest request) {
+        Game game = getGameFromRoomId(roomId);
+
+        if (game.getGameTimer() == null || !game.getGameTimer().isTimeUp()) {
+            throw new ApiException(GameError.GAME_NOT_OVER);
+        }
```

Comment:
When we add #96, will we add `allSolved` to the if statement?

Reply:
Yeah probably (maybe we could add an isGameOver helper method or even a gameOver variable to Game.java assuming in the future the end game criteria might grow more complicated)

Another note is after #96 is merged I'll also update the Game.tsx file to pass in the correct objects to the results page